extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -32,61 +31,78 @@ public DevConsoleRuntimeTemplateInfoBuildItem collectScores() {
             Path resource = i.getChildPath(StaticResourcesRecorder.META_INF_RESOURCES);
             if (resource != null && Files.exists(resource)) {
-                collectKnownPaths(resource, knownPaths);
+                collectKnownPaths(resource, staticResourceInfo);
             }
         }
 
         ClassPathUtils.consumeAsPaths(StaticResourcesRecorder.META_INF_RESOURCES, resource -> {
-            collectKnownPaths(resource, knownPaths);
+            collectKnownPaths(resource, staticResourceInfo);
         });
-
-        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", knownPaths);
+        return new DevConsoleTemplateInfoBuildItem("staticResourceInfo", staticResourceInfo);
     }
 
-    private void collectKnownPaths(Path resource, Set<String> staticResources) {
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
+                final Stack<StaticResourceInfo.StaticFile> currentFolder = new Stack<>();
 
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
-                        } else {
-                            String parentString = parent.toString();
-                            if (!parentString.startsWith("/")) {
-                                parentString = "/" + parentString;
-                            }
-                            staticResources.add(parentString + "/");
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+
+                    Path relativeFile = resource.relativize(dir);
+                    String relativeFileString = "/" + relativeFile.toString();
+                    Path dirName = dir.getFileName();
+                    StaticResourceInfo.StaticFile getStaticFile = staticResourceInfo.resourceMap.get(relativeFileString);
+
+                    if (getStaticFile == null) {
+                        getStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(), true);
+                        staticResourceInfo.resourceMap.put(relativeFileString, getStaticFile);
+
+                        // adding to parent
+                        if (!currentFolder.isEmpty()) {
+                            currentFolder.peek().children.add(getStaticFile);
                         }
+                        currentFolder.push(getStaticFile);
```

Comment:
You need to push it even if you didn't create it, otherwise you're skipping a level and its children will be put in the wrong parent.

Reply:
Actually, if I am pushing it even when I didn't create then I am getting the `file` twice.

Something like this : 
![image](https://user-images.githubusercontent.com/46818757/107904030-b9d9a180-6f70-11eb-8ea4-926afd32b97c.png)

When I push only inside `if` i.e only when I am creating it then it works fine :)